x-pack/plugins/ml/public/maps/anomaly_layer_wizard_factory.tsx

```diff
@@ -45,18 +46,23 @@ export class AnomalyLayerWizardFactory {
     this.canGetJobs = canGetJobs;
   }
 
-  private async getServices(): Promise<{ mlJobsService: MlApiServices['jobs'] }> {
-    const [coreStart] = await this.getStartServices();
+  private async getServices(): Promise<{ mlJobsService: MlApiServices['jobs']; mlLocator: any }> {
+    const [coreStart, pluginStart] = await this.getStartServices();
     const { jobsApiProvider } = await import('../application/services/ml_api_service/jobs');
 
     const httpService = new HttpService(coreStart.http);
     const mlJobsService = jobsApiProvider(httpService);
+    const mlLocator = pluginStart.share.url.locators.get(ML_APP_LOCATOR);
 
-    return { mlJobsService };
+    return { mlJobsService, mlLocator };
   }
 
   public async create(): Promise<LayerWizard> {
-    const { mlJobsService } = await this.getServices();
+    const { mlJobsService, mlLocator } = await this.getServices();
+    const jobsManagementPath = await mlLocator.getUrl({
```

Comment:
Since locators.get() can return undefined, would be good to handle the scenario for when mlLocator is undefined. Suggestion to console.error/throw error if that's the case.

Also should we be checking if user has permission to create jobs or not before showing the `Create jobs` prompt?

Reply:
Good call - added `canCreateJob` check and disabled the creation prompt if user doesn't have creation permission in cb6ff5f3e2f81603a2a34444a7f777206e504896

Also added undefined mlLocator check in this change. Now the user will only see the empty prompt if the locator is defined.